### ISSUE TITLE
SCM: yaml whitespace error on re-import

### ISSUE THREAD

**Issue by LordMike** [rundeck/rundeck#1496]
I just comitted a job with an embedded script, and did the re-import thing. Now suddenly the job changed again. The diff shows the following:

![image](https://cloud.githubusercontent.com/assets/1027111/10594004/d3214950-76c8-11e5-90f6-f66be3ce654d.png)

Removing trailing whitespaces can be dangerous. I'm off to bed, so I can't investigate until tomorrow evening.

**Comment by gschueler** [comment]
Is this a line-ending issue? to me it looks like the line endings changed, and red whitespace warning you see may be spurious (yaml is indenting the text for you but that indentation will be removed when it is parsed).

We encountered a line-ending issue with XML, e.g. if you enter multiple lines in a Job description, it is serialized inside an XML CDATA section to preserve the whitespace... however the value stored in the DB includes "CRLF" line endings as are sent by the web browser, while sometimes the serialization alters it to "LF".   We have altered it to always serialize XML with Unix line endings (LF aka `\n`).

For YAML we may have to do the same thing.  Whether it uses Unix, Dos, or native platform line endings could be a configuration setting.
